force-app/test/utilities/test-environment.js

```diff
@@ -47,6 +50,13 @@ export class TestEnvironment {
         this.#username = this.#getKeyFromEnv('username');
         this.#password = this.#getKeyFromEnv('password');
         this.#sfdxLoginUrl = this.#getKeyFromEnv('loginUrl');
+
+        // Halt process if Salesforce session timed out
+        const loginTimestamp = this.#getKeyFromEnv('loginTimestamp');
+        if (!loginTimestamp || new Date().getTime() - parseInt(loginTimestamp, 10) > SESSION_TIMEOUT) {
+            console.error('FATAL ERROR: Salesforce session timed out. Re-authenticate before running tests.');
```

Comment:
nit: I would use something a bit less scary than `FATAL ERROR` for the error label as we aren't in a critical context (i.e kernel)

Reply:
Ok, downgrading this to just ERROR.